src/Components/AccountPageSubPages/ChangePassword.jsx

```diff
@@ -1,3 +1,98 @@
-export default function ChangePassword() {
-  return <h1>Fill in the below form to change your password</h1>;
+import { useFormik } from "formik";
+import { useState } from "react";
+import { changePassword } from "../../Networking";
+import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
+
+export default function ChangePassword(props) {
+  const { changeTab } = props;
+  const [isIncorrectPassword, setisIncorrectPassword] = useState(false);
+
+  function validate(values) {
+    const errors = {};
+    if (!values.username) {
+      errors.username = "Required";
+    }
+    if (!values.oldPassword) {
+      errors.oldPassword = "Required";
+    }
+    if (!values.newPassword) {
+      errors.newPassword = "Required";
+    }
+    return errors;
+  }
+
+  const { handleSubmit, handleChange, handleBlur, touched, errors } = useFormik(
+    {
+      initialValues: {
+        username: "",
+        oldPassword: "",
+        newPassword: "",
+      },
+      validate,
+      onSubmit: async (values) => {
+        const response = await changePassword(
+          values.username,
+          values.oldPassword,
+          values.newPassword
+        );
+        if (response.code === 200) {
+          changeTab("leaderboard");
+          alert("password changed successfully");
```

Comment:
what happens when old password = new password? maybe another error message to account for that